### DISCUSSION TITLE
Discrepancy between not-found page Behavior in Development and Production Environments

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#50849] [Help]
### Summary

I am currently working on a new website with the App Router. However, I am facing a problem with the not found page.

When I start a local server in a development environment, and call a path in the browser that does not exist, the page I defined in `not-found.tsx` appears.

However, when I deploy the project with vercel in a production environment, and call the same route (non-existing page) that previously referenced the notfound page I defined, I am shown the default page defined by nextjs. 

Do you have any idea what could be the reason for this? I appreciate any feedback and criticism on my issue! Please let me know, if you need further information about my project setup/structure.

Link to existing path:
[orbitgrowth.de/about](https://orbitgrowth.de/about)

Link to non-existing path:
[orbitgrowth.de/contact](https://orbitgrowth.de/contact)

### Additional information

```js
yarn next info returns:

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home  
    Binaries:
      Node: 16.15.1
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.4.3
      eslint-config-next: 13.4.2
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

**Comment by icyJoseph** [comment]
I've seen reports for this in the issues tab also. However, two projects I have deployed, with `not-found` work correctly, for example:

- https://next-js-react-ct9wkgsaq-icyjoseph.vercel.app/foo (app dir not-found)

I know it is correct, because the pages dir 404 page used styled-components, and the app dir version uses CSS modules. For reference:

- https://next-js-react-gbg.vercel.app/foo (pages dir 404)

I wonder if you have also `pages/_error` setup? Perhaps that's causing conflicts in the prod build? Have you verified this locally, by doing `next build && next start`? Got a repo to look at?

**Reply by ghost** [reply]
Sure, here is a link to the repository:

[github.com/clangerOg/orbit-growth-web](https://github.com/clangerOg/orbit-growth-web)

I have already tested the project several times locally and so far everything worked fine, even in production mode (`next build && next start`).

I don't use the `pages` directory in any way in my project at the moment, but I expect to use it for form submissions in the future.
<br />
(In the following I'm talking in the context of production mode)
When I call non-existent pages locally, the page I defined appears. However, I noticed that when I call the path `/_error` I get an unstyled page in the browser with "internal server error" and the console gives the following error message:

```cmd
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './server.edge' is not defined by "exports" in G:\organizations\orbit-growth\orbit-growth-web\node_modules\react-dom\package.json
    at new NodeError (node:internal/errors:372:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:472:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:753:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at G:\organizations\orbit-growth\orbit-growth-web\node_modules\next\dist\server\require-hook.js:180:36
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

**Reply by icyJoseph** [reply]
I think it is best to create an issue or report on this one https://github.com/vercel/next.js/issues/50590
